back/app/controllers/web_api/v1/users_controller.rb

```diff
@@ -245,6 +245,7 @@ def reset_confirm_on_existing_no_password_user?
     return false unless errors.any? { |hash| hash[:error] == :taken }
 
     existing_user = User.find_by(email: @user.email)
+    return false unless existing_user
```

Comment:
To be honest, I don't know how `email: :taken` can happen at the same time with `nil` from `find_by(email:`. But this change will at least give our users some hint of what's wrong (by returning a JSON error instead of 500) and will stop adding errors in Sentry.